pyproject.toml

```diff
@@ -1,3 +1,150 @@
 [tool.black]
 line-length = 100
 include = '\.pyi?$'
+
+[tool.poetry]
+name = "pennylane"
+version = "0.35.0-dev"
+description = "PennyLane is a Python quantum machine learning library by Xanadu Inc."
+authors = ["PennyLaneAI <info@pennylane.ai>"]
+license = "Apache-2.0"
+readme = "README.md"
+homepage = "https://pennylane.ai/"
+repository = "https://github.com/PennyLaneAI/pennylane"
+
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
```

Comment:
We shouldn't be converting the PEP-517 complaint build system here to be using Poetry by default.

Reply:
as mentioned on your other comment, I'm not sure if it's the wrong thing to do. Poetry [claims it's standard](https://python-poetry.org/docs/pyproject/#poetry-and-pep-517), but I'll admit that I don't know what other 517-compliant packages look like